pkg/server/testing/fake_execution_client.go

```diff
@@ -89,6 +89,7 @@ func (f *FakeExecutionClient) Stop() {
 	for _, client := range f.eventClients {
 		close(client.Events)
 	}
+	f.eventClients = nil
```

Comment:
isn't the underlying root issue here that f.eventsQueue isn't thread safe? 

Did you want to kill eventClients or empty it?  `f.eventClients = f.eventClients[:0]`